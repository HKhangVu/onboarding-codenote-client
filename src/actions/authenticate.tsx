export const LOG_IN = "LOG_IN";

export function userHasAuthenticated (val) {
    return {
      type: LOG_IN,
      isLogin: val
    }
}