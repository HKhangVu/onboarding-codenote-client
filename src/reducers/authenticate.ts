const initialState = {
  isAuthenticated: false,
};

export default function authenticate(state = initialState, action) { 
  return state;
};
