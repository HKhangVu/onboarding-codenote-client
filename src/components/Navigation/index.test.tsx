import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux';
import { NavItem} from "react-bootstrap";
import { History } from 'history';
import Navigation from '../../components/Navigation';
import store from '../../store';
import { userHasAuthenticated } from '../../actions/authenticate';

describe("Navigation Bar", () => {
  test('render after login', () => {
    store.dispatch(userHasAuthenticated(true));
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter >
          <Navigation history={new Object as History} isAuthenticated userHasAuthenticated={(val:true)=>{}} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('render without login', () => {
    store.dispatch(userHasAuthenticated(false));
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter >
          <Navigation history={new Object as History} isAuthenticated userHasAuthenticated={(val:true)=>{}}/>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
    
  test('logout button', ()  => {
    let iprop =  {history : ({push:()=>{}} as any) as History,isAuthenticated:true,userHasAuthenticated:()=>{}}
    store.dispatch(userHasAuthenticated(true));
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter >
          <Navigation {...iprop} />
        </MemoryRouter>
      </Provider>
    ); 
    const navInstance = wrapper.find(NavItem);
    navInstance.props().onClick = jest.fn();
    wrapper.instance().forceUpdate();
    const logoutBtn = wrapper.find(NavItem).find('a');
    navInstance.props().onClick();
    // navInstance.instance().handleLogout();
    // logoutBtn.props().onClick();
    expect( navInstance.props().onClick).toHaveBeenCalled();
  });
});
