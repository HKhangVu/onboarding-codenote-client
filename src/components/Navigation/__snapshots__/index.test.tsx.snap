// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar render after login 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "6ayk4s",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "6ayk4s",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Navigation)
        history={Object {}}
        isAuthenticated={true}
        userHasAuthenticated={[Function]}
      >
        <Navigation
          history={Object {}}
          isAuthenticated={true}
          userHasAuthenticated={[Function]}
        >
          <Uncontrolled(Navbar)
            bsStyle="default"
            collapseOnSelect={true}
            fluid={true}
          >
            <Navbar
              bsClass="navbar"
              bsStyle="default"
              collapseOnSelect={true}
              componentClass="nav"
              fixedBottom={false}
              fixedTop={false}
              fluid={true}
              inverse={false}
              onToggle={[Function]}
              staticTop={false}
            >
              <nav
                className="navbar navbar-default"
              >
                <Grid
                  bsClass="container"
                  componentClass="div"
                  fluid={true}
                >
                  <div
                    className="container-fluid"
                  >
                    <NavbarHeader>
                      <div
                        className="navbar-header"
                      >
                        <NavbarBrand>
                          <Link
                            className="navbar-brand"
                            replace={false}
                            to="/"
                          >
                            <a
                              className="navbar-brand"
                              href="/"
                              onClick={[Function]}
                            >
                              CodeNote
                            </a>
                          </Link>
                        </NavbarBrand>
                        <NavbarToggle>
                          <button
                            className="navbar-toggle collapsed"
                            onClick={[Function]}
                            type="button"
                          >
                            <span
                              className="sr-only"
                            >
                              Toggle navigation
                            </span>
                            <span
                              className="icon-bar"
                            />
                            <span
                              className="icon-bar"
                            />
                            <span
                              className="icon-bar"
                            />
                          </button>
                        </NavbarToggle>
                      </div>
                    </NavbarHeader>
                    <NavbarCollapse>
                      <Collapse
                        appear={false}
                        dimension="height"
                        getDimensionValue={[Function]}
                        in={false}
                        mountOnEnter={false}
                        timeout={300}
                        unmountOnExit={false}
                      >
                        <Transition
                          appear={false}
                          aria-expanded={null}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <div
                            aria-expanded={null}
                            className="navbar-collapse collapse"
                          >
                            <Nav
                              bsClass="nav"
                              justified={false}
                              pullLeft={false}
                              pullRight={true}
                              stacked={false}
                            >
                              <ul
                                className="nav navbar-nav navbar-right"
                                role={null}
                              >
                                <NavItem
                                  active={false}
                                  disabled={false}
                                  key=".0"
                                  onClick={[Function]}
                                  onSelect={[Function]}
                                >
                                  <li
                                    className=""
                                    role="presentation"
                                  >
                                    <SafeAnchor
                                      componentClass="a"
                                      disabled={false}
                                      onClick={[Function]}
                                    >
                                      <a
                                        href="#"
                                        onClick={[Function]}
                                        onKeyDown={[Function]}
                                        role="button"
                                      >
                                        Logout
                                      </a>
                                    </SafeAnchor>
                                  </li>
                                </NavItem>
                              </ul>
                            </Nav>
                          </div>
                        </Transition>
                      </Collapse>
                    </NavbarCollapse>
                  </div>
                </Grid>
              </nav>
            </Navbar>
          </Uncontrolled(Navbar)>
        </Navigation>
      </Connect(Navigation)>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`Navigation Bar render without login 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "vv0kor",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "vv0kor",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Navigation)
        history={Object {}}
        isAuthenticated={true}
        userHasAuthenticated={[Function]}
      >
        <Navigation
          history={Object {}}
          isAuthenticated={false}
          userHasAuthenticated={[Function]}
        >
          <Uncontrolled(Navbar)
            bsStyle="default"
            collapseOnSelect={true}
            fluid={true}
          >
            <Navbar
              bsClass="navbar"
              bsStyle="default"
              collapseOnSelect={true}
              componentClass="nav"
              fixedBottom={false}
              fixedTop={false}
              fluid={true}
              inverse={false}
              onToggle={[Function]}
              staticTop={false}
            >
              <nav
                className="navbar navbar-default"
              >
                <Grid
                  bsClass="container"
                  componentClass="div"
                  fluid={true}
                >
                  <div
                    className="container-fluid"
                  >
                    <NavbarHeader>
                      <div
                        className="navbar-header"
                      >
                        <NavbarBrand>
                          <Link
                            className="navbar-brand"
                            replace={false}
                            to="/"
                          >
                            <a
                              className="navbar-brand"
                              href="/"
                              onClick={[Function]}
                            >
                              CodeNote
                            </a>
                          </Link>
                        </NavbarBrand>
                        <NavbarToggle>
                          <button
                            className="navbar-toggle collapsed"
                            onClick={[Function]}
                            type="button"
                          >
                            <span
                              className="sr-only"
                            >
                              Toggle navigation
                            </span>
                            <span
                              className="icon-bar"
                            />
                            <span
                              className="icon-bar"
                            />
                            <span
                              className="icon-bar"
                            />
                          </button>
                        </NavbarToggle>
                      </div>
                    </NavbarHeader>
                    <NavbarCollapse>
                      <Collapse
                        appear={false}
                        dimension="height"
                        getDimensionValue={[Function]}
                        in={false}
                        mountOnEnter={false}
                        timeout={300}
                        unmountOnExit={false}
                      >
                        <Transition
                          appear={false}
                          aria-expanded={null}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <div
                            aria-expanded={null}
                            className="navbar-collapse collapse"
                          >
                            <Nav
                              bsClass="nav"
                              justified={false}
                              pullLeft={false}
                              pullRight={true}
                              stacked={false}
                            >
                              <ul
                                className="nav navbar-nav navbar-right"
                                role={null}
                              >
                                <LinkContainer
                                  activeClassName="active"
                                  exact={false}
                                  replace={false}
                                  strict={false}
                                  to="/signup"
                                >
                                  <Route
                                    exact={false}
                                    path="/signup"
                                    strict={false}
                                  >
                                    <NavItem
                                      active={false}
                                      className=""
                                      disabled={false}
                                      href="/signup"
                                      onClick={[Function]}
                                    >
                                      <li
                                        className=""
                                        role="presentation"
                                      >
                                        <SafeAnchor
                                          componentClass="a"
                                          disabled={false}
                                          href="/signup"
                                          onClick={[Function]}
                                        >
                                          <a
                                            href="/signup"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                          >
                                            Signup
                                          </a>
                                        </SafeAnchor>
                                      </li>
                                    </NavItem>
                                  </Route>
                                </LinkContainer>
                                <LinkContainer
                                  activeClassName="active"
                                  exact={false}
                                  replace={false}
                                  strict={false}
                                  to="/login"
                                >
                                  <Route
                                    exact={false}
                                    path="/login"
                                    strict={false}
                                  >
                                    <NavItem
                                      active={false}
                                      className=""
                                      disabled={false}
                                      href="/login"
                                      onClick={[Function]}
                                    >
                                      <li
                                        className=""
                                        role="presentation"
                                      >
                                        <SafeAnchor
                                          componentClass="a"
                                          disabled={false}
                                          href="/login"
                                          onClick={[Function]}
                                        >
                                          <a
                                            href="/login"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                          >
                                            Login
                                          </a>
                                        </SafeAnchor>
                                      </li>
                                    </NavItem>
                                  </Route>
                                </LinkContainer>
                              </ul>
                            </Nav>
                          </div>
                        </Transition>
                      </Collapse>
                    </NavbarCollapse>
                  </div>
                </Grid>
              </nav>
            </Navbar>
          </Uncontrolled(Navbar)>
        </Navigation>
      </Connect(Navigation)>
    </Router>
  </MemoryRouter>
</Provider>
`;
